imsim/photon_ops.py

```diff
@@ -135,12 +119,248 @@ def applyTo(self, photon_array, local_wcs=None, rng=None):
         photon_array.y -= self.image_pos.y
 
     def __str__(self):
-        return "imsim.LsstPhotonOp()"
+        return f"imsim.{type(self).__name__}()"
+
+    def __repr__(self):
+        return str(self)
+
+
+class LsstDiffraction(PhotonOp):
+    """Photon operator that applies statistical diffraction by the
+    LSST spider.
+
+    Parameters
+    ----------
+    telescope : batoid.Optic
+        The telescope to trace through.
+    latitude : float
+        Geographic latitude of telescope. Needed to calculate the field rotation
+        for the spider diffraction.
+    altitude, azimuth : float
+        alt/az coordinates the telescope is pointing to (in degree).
+    sky_pos : galsim.CelestialCoord
+    icrf_to_field : galsim.GSFitsWCS
+    shift_optics : dict[str, list[float]]
+        A dict mapping optics keys to shifts represented by a list of 3 floats.
+        The corresponding optics will be displaced by the specified corrdinates.
+        Example config for perturbed+defocused telescope to obtain a donut:
+        -
+            type: lsst_optics
+            ...
+            shift_optics:
+              Detector: [0, 0, 1.5e-3]
+              M2: [3.0e-3, 0, 0]
+    seed : An optional seed forwarded to the angle distribution used
+           for diffraction.
+    """
+
+    _req_params = {
+        "telescope": Optic,
+        "band": str,
+        "latitude": float,
+    }
+    _opt_params = {"shift_optics": dict}
+
+    def __init__(
+        self,
+        telescope,
+        latitude,
+        altitude,
+        azimuth,
+        sky_pos,
+        icrf_to_field,
+        shift_optics=None,
+        seed=None,
+    ):
+        if shift_optics is not None:
+            for optics_key, shift in shift_optics.items():
+                telescope = telescope.withGloballyShiftedOptic(optics_key, shift)
+        self.telescope = telescope
+        self.latitude = latitude
+        self.altitude = altitude
+        self.azimuth = azimuth
+        self.sky_pos = sky_pos
+        self.icrf_to_field = icrf_to_field
+
+        deviate = GaussianDeviate(seed=seed)
```

Comment:
This isn't right.  We need to be using the config rng, not making a new one with some random seed (especially not default seed=None).  Basically, the whole diffraction_rng stuff needs to be in `applyTo` and use the `rng` paramter there.  And remove the `seed` parameter here.